persistent-test/src/ForeignKey.hs

```diff
@@ -33,6 +42,45 @@ share [mkPersist persistSettings { mpsGeneric = False }, mkMigrate "compositeMig
     Primary name
     Foreign SelfReferenced OnDeleteCascade fkparent pname
     deriving Show Eq
+
+  A
+    aa String
+    ab Int
+    U1 aa
+
+  B
+    ba String
+    bb Int
+    Foreign A OnDeleteCascade fkA ba References aa
+    deriving Show Eq
+
+  AComposite
+    aa String
+    ab Int
+    U2 aa ab
+
+  BComposite
+    ba String
+    bb Int
+    Foreign AComposite OnDeleteCascade fkAComposite ba bb References aa ab
+    deriving Show Eq
+
+  BExplicit
+    ba AId noreference
+    Foreign A OnDeleteCascade fkAI ba References Id
+    deriving Show Eq
+
+  Chain
+    name String
+    previous ChainId Maybe noreference
+    Foreign Chain OnDeleteSetNull fkChain previous References Id
+    deriving Show Eq
+
+  Chain2
+    name String
+    previous Chain2Id Maybe noreference
```

Comment:
Okay, so we're still using `noreference` as a work-around to get the cascades working I take it